examples/nft-nd/nft.go

```diff
@@ -285,3 +287,23 @@ func Properties(id []byte) map[string]string {
 	}
 	return result
 }
+
+// RoyaltyRecipient contains information about the recipient and the royalty amount.
+type RoyaltyRecipient struct {
+	Address interop.Hash160
+	Amount  *big.Int
+}
+
+func RoyaltyInfo(tokenID []byte, royaltyToken interop.Hash160, salePrice int) []RoyaltyRecipient {
+	ctx := storage.GetReadOnlyContext()
+	owner := getOwnerOf(ctx, tokenID)
+	if !runtime.CheckWitness(owner) {
+		panic("invalid owner")
+	}
+	return []RoyaltyRecipient{
```

Comment:
Read [method spec](https://github.com/neo-project/proposals/blob/master/nep-24.mediawiki#common-methods) one more time, rewrite this method according to the spec. It should be available for any caller. Also, the `royaltyToken` itself must be checked because it may be some side token not emitted by this contract.

Reply:
See the reference example: https://github.com/Jim8y/neo-devpack-dotnet/blob/6bbb5fb461f0a4cc0de177fa5f47db4c9f185015/src/Neo.SmartContract.Framework/Interfaces/INEP24.cs#L34